.github/workflows/ci-cd.yml

```diff
@@ -114,6 +114,11 @@ jobs:
     outputs:
       released: ${{ steps.release.outputs.released }}
 
+    concurrency: release
+    permissions:
+      id-token: write
```

Comment:
why is this privilege needed?

Reply:
python-semantic-release/python-semantic-release fails without it

https://github.com/aio-libs/aiohappyeyeballs/actions/runs/10184595236/job/28172458633

Reply:
@bdraco that's because of `contents: write`. OIDC is much more dangerous and shouldn't be added lightly.

Reply:
https://python-semantic-release.readthedocs.io/en/latest/github-action.html

Reply:
It looks like https://github.com/python-semantic-release/python-semantic-release/blob/4e59b501374fe6c6f822cc0e9eb41da3e896f642/action.yml#L62 has some options that could be turned off

`commit`, `tag`, `push`, `changelog`

I'm not sure if its better to move them to another step or do it here

Reply:
That doesn't explain `id-token`. I know the following job uses it for trusted publishing, but this one probably shouldn't.